includes/Dashboard.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Dashboard class.
+ *
+ * Responsible for adding the stories dashboard to WordPress admin.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+/**
+ * Dashboard class.
+ */
+class Dashboard {
+	/**
+	 * Script handle.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_HANDLE = 'stories-dashboard';
+
+	/**
+	 * Admin page hook suffix.
+	 *
+	 * @var string|false The dashboard page's hook_suffix, or false if the user does not have the capability required.
+	 */
+	private $hook_suffix;
+
+	/**
+	 * Initializes the dashboard logic.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_action( 'admin_menu', [ $this, 'add_menu_page' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+	}
+
+	/**
+	 * Returns the admin page's hook suffix.
+	 *
+	 * @return string|false The dashboard page's hook_suffix, or false if the user does not have the capability required.
+	 */
+	public function get_hook_suffix() {
+		return $this->hook_suffix;
+	}
+
+	/**
+	 * Registers the dashboard admin menu page.
+	 *
+	 * @return void
+	 */
+	public function add_menu_page() {
+		$this->hook_suffix = add_submenu_page(
+			'edit.php?post_type=' . Story_Post_Type::POST_TYPE_SLUG,
+			__( 'Dashboard', 'web-stories' ),
+			__( 'Dashboard', 'web-stories' ),
+			'edit_posts',
+			'stories-dashboard',
+			[ $this, 'render' ],
+			0
+		);
+	}
+
+	/**
+	 * Renders the dashboard page.
+	 *
+	 * @return void
+	 */
+	public function render() {
+		?>
+		<div id="web-stories-dashboard">
+			<h1 class="loading-message"><?php esc_html_e( 'Please wait...', 'web-stories' ); ?></h1>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Enqueues dashboard scripts and styles.
+	 *
+	 * @param string $hook_suffix The current admin page.
+	 *
+	 * @return void
+	 */
+	public function enqueue_assets( $hook_suffix ) {
+		if ( $this->hook_suffix !== $hook_suffix ) {
+			return;
+		}
+
+		$asset_file   = WEBSTORIES_PLUGIN_DIR_PATH . 'assets/js/' . self::SCRIPT_HANDLE . '.asset.php';
```

Comment:
Just curious -- is the logic of keeping the related PHP files in JS folder based on Gutenberg or is this something that seems to be generally recommended approach?

Reply:
It's something Gutenberg has spearheaded with its `@wordpress/dependency-extraction-webpack-plugin` package, which is used by Gutenberg, WordPress core, the AMP plugin, and this one too.

I know that the location is not ideal and can be perceived as odd, but at the moment it is not easily changeable (follow https://github.com/WordPress/gutenberg/issues/20562 for updates).